frontend/packages/kubevirt-plugin/src/components/modals/menu-actions-modals/vmi-users-alert.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import { Alert, Button } from '@patternfly/react-core';
+import { history } from '@console/internal/components/utils';
+import { getName, getNamespace } from '@console/shared/src/selectors/common';
+import { useGuestAgentInfo } from '../../../hooks/use-guest-agent-info';
+import { GuestAgentInfoWrapper } from '../../../k8s/wrapper/vm/guest-agent-info/guest-agent-info-wrapper';
+import { VMIKind } from '../../../types';
+
+import './vmi-users-alert.scss';
+
+type VMIUsersAlertProps = {
+  vmi: VMIKind;
+  alertTitle?: string;
+  cancel: () => void;
+};
+
+export const VMIUsersAlert: React.FC<VMIUsersAlertProps> = ({ vmi, cancel, alertTitle }) => {
+  const [guestAgentInfoRaw] = useGuestAgentInfo({ vmi });
+  const guestAgentInfo = new GuestAgentInfoWrapper(guestAgentInfoRaw);
+  const userListLength = guestAgentInfo.getUserList()?.length || 0;
+
+  const name = getName(vmi);
+  const namespace = getNamespace(vmi);
+  const alertHref = `/k8s/ns/${namespace}/virtualmachines/${name}/details#logged-in-users`;
+  const onLinkClick = () => {
+    cancel();
+    history.push(alertHref);
+  };
+
+  let alertBody = null;
+  if (userListLength === 1) {
+    alertBody = (
+      <>
+        <Button variant="link" isInline onClick={onLinkClick}>
+          One user
+        </Button>{' '}
+        is currently logged in to this VM. Proceeding with this operation may cause logged in users
+        data loss.
+      </>
+    );
+  } else if (userListLength > 1) {
+    alertBody = (
```

Comment:
this feels like it should be parametrized

Reply:
Agreed. I think this can be done in one `Button` component using `pluralize`. I don't think it will hurt anything if we use a numeral 1 instead of "One".

Reply:
+2 :-) fixing

Reply:
fixed